state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -376,6 +380,15 @@ pub mod pallet {
 		pub boost_fee: BasisPoints,
 	}
 
+	#[derive(
+		CloneNoBound, RuntimeDebugNoBound, PartialEqNoBound, EqNoBound, Encode, Decode, TypeInfo,
+	)]
+	#[scale_info(skip_type_params(T, I))]
+	pub enum DepositFailedDetails<T: Config<I>, I: 'static> {
```

Comment:
I'm a little concerned about exposing *everything* we receive in an event. Feels like we should be more intentional about what we put in the public interface. I see the rationale for providing as much info as possible, but this also means we are more likely break compatibility (either intentionally or not) whenever these mostly internal structs are modified (I know CFEs also uses them, but could still be considered internal).

As a thought, maybe there is a way to separate fields that can be dependant upon externally from "extra info" that's just for informational purposes (i.e. it could be dumped somewhere in a text field in the explorer). Not sure if the consumers of these events will be disciplined enough to not start depending on that info as well.

Interested to hear your thoughts @dandanlen.

Reply:
> Not sure if we (or other consumers of these events) will be disciplined enough to not start depending on that info as well.

Assume whatever is public will be used by someone eventually :)

Reply:
> > Not sure if we (or other consumers of these events) will be disciplined enough to not start depending on that info as well.
> 
> Assume whatever is public will be used by someone eventually :)

Yes, this essentially: https://www.hyrumslaw.com/

Reply:
Didn't know that one, thanks! It's very correct indeed :)

Reply:
While I tend to agree, I also feel like this can reasonably be considered an exception to this rule. With errors we do want to provide as much context as possible, and errors should not be (and generally are not) used for 'control flow' - only for display. (like you mentioned - just 'dump it somewhere' for debugging)

In this case, I feel like it's reasonable to consider the action/intent of the failed operation to be useful context. Maybe we can be less vague about the purpose of this field and just call it 'error context' or 'metadata' or something.

I'm open to being more deliberate and only exposing the necessary context as long as it's obvious what that necessary context is. For example, as an alternative, we could use a richer type for the error reason, that includes the value that caused the issue? 

To recap - where this all came from was that we wanted access to the destination asset and the transaction_in_id in DepositIgnored and CcmFailed events. Then we realised that there is lot of duplicate information, so we merged the two events and added the extra context for vault swaps. I feel like the new approach is more consistent and structured than what we had previously.